quarkus/config-api/src/main/java/org/keycloak/config/ProxyOptions.java

```diff
@@ -3,10 +3,24 @@
 public class ProxyOptions {
 
     public enum Mode {
-        none,
+        none(false),
+        passthrough(false),
         edge,
-        reencrypt,
-        passthrough;
+        reencrypt;
+
+        private final boolean proxyEnabled;
+
+        Mode(boolean proxyEnabled) {
+            this.proxyEnabled = proxyEnabled;
+        }
+
+        Mode() {
+            this(true);
+        }
+
+        public boolean isProxyEnabled() {
```

Comment:
I'd rename this to `isProxyHeadersEnabled` as `passthrough` is still a proxy mode and e.g. should use 443 port.